test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs

```diff
@@ -30,210 +28,80 @@ public ResponseCompressionTests(ITestOutputHelper output) : base(output)
         {
         }
 
-        // IIS Express
-        [ConditionalTheory]
-        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Portable, HostingModel.OutOfProcess, ANCMVersion.AspNetCoreModule)]
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Portable, HostingModel.OutOfProcess, ANCMVersion.AspNetCoreModuleV2)]
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Portable, HostingModel.InProcess, ANCMVersion.AspNetCoreModuleV2)]
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Standalone, HostingModel.OutOfProcess, ANCMVersion.AspNetCoreModule)]
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Standalone, HostingModel.OutOfProcess, ANCMVersion.AspNetCoreModuleV2)]        
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Standalone, HostingModel.InProcess, ANCMVersion.AspNetCoreModuleV2)]
-        [InlineData(RuntimeFlavor.Clr, ApplicationType.Standalone, HostingModel.OutOfProcess, ANCMVersion.AspNetCoreModule, Skip = "Websdk issue with full framework publish. See https://github.com/aspnet/websdk/pull/322")]
-        [InlineData(RuntimeFlavor.Clr, ApplicationType.Standalone, HostingModel.OutOfProcess, ANCMVersion.AspNetCoreModuleV2, Skip = "Websdk issue with full framework publish. See https://github.com/aspnet/websdk/pull/322")]
-        public Task ResponseCompression_IISExpress_NoCompression(RuntimeFlavor runtimeFlavor, ApplicationType applicationType, HostingModel hostingModel, ANCMVersion ancmVersion)
-        {
-            return ResponseCompression(ServerType.IISExpress, 
-                runtimeFlavor, 
-                RuntimeArchitecture.x64, 
-                CheckNoCompressionAsync, 
-                applicationType,
-                hostCompression: false, 
-                hostingModel: hostingModel,
-                ancmVersion: ancmVersion);
-        }
+        public static TestList NoCompressionTestVariants
+            => TestMatrix.ForServers(ServerType.IISExpress, ServerType.Kestrel, ServerType.Nginx, ServerType.HttpSys)
+                .WithTfms(Tfm.NetCoreApp22, Tfm.Net461)
+                .WithAllAncmVersions()
```

Comment:
Can we make this a default if IIS is used and versions and the hosting model wasn't set explicitly?

Reply:
I tried to be really conservative with the defaults because we don't want to run more tests than we actually need. I know we don't want both versions by default for IISIntegration tests, I'm not sure about elsewhere.

Reply:
For the places where we know what we want we would set them, but all version all models seems sensible by default.